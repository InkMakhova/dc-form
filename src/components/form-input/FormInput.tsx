import { JSX } from 'react'
import { Input, Label, TextField } from 'react-aria-components'
import styles from './FormInput.module.css'
import { FormInputProps } from './FormInput.types.ts'

function FormInput({label, placeholder, size, name, disabled}: FormInputProps): JSX.Element {
  return (
    <TextField className={styles['input__container']} isDisabled={disabled}>
      <Label>{label}</Label>
      <Input
        placeholder={placeholder}
        name={name}
        disabled={disabled}
        className={`${styles['input__field']} ${styles[`input__field--${size}`]}`}
      />
    </TextField>
  )
}

export default FormInput;
