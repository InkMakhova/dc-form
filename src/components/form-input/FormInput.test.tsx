import { render, screen } from '@testing-library/react'
import { expect } from 'vitest'
import { InputSize } from './FormInput.types.ts'
import FormInput from './FormInput.tsx'

test('Input has correct name attribute', () => {
  render(
    <FormInput
      name="custom-name"
      label="Name Field"
      placeholder="Name"
      size={InputSize.Medium}
    />
  )
  const input = screen.getByLabelText('Name Field');
  expect(input).toHaveAttribute('name');
});

test('Input renders with label', () => {
  render(
    <FormInput
      name="medium-input"
      label="Label"
      placeholder="enter text"
      size={InputSize.Medium}
    />
  )
  expect(screen.getByLabelText("Label")).toBeInTheDocument();
});

test('Input applies correct classes for size', () => {
  render(
    <FormInput
      name="medium-input"
      label="Label"
      placeholder="enter text"
      size={InputSize.Medium}
    />
  )
  const input = screen.getByLabelText('Label');
  expect(input.className).toMatch(/input__field--medium/);
});

test('Input displays correct placeholder', () => {
  render(
    <FormInput
      name="medium-input"
      label="Label"
      placeholder="enter text"
      size={InputSize.Medium}
    />
  )
  expect(screen.getByPlaceholderText('enter text')).toBeInTheDocument();
});

test('Input is disabled when disabled prop is true', () => {
  render(
    <FormInput
      name="disabled-input"
      label="Disabled"
      placeholder="cannot type"
      size={InputSize.Small}
      disabled
    />
  )
  const input = screen.getByLabelText('Disabled');
  expect(input).toBeDisabled();
});

test('matches snapshot', () => {
  const { container } = render(
    <FormInput
      name="snapshot"
      label="Label"
      placeholder="Text"
      size={InputSize.Medium}
    />
  )
  expect(container).toMatchSnapshot();
});
