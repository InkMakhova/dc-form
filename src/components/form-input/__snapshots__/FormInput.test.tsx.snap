// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="_container_7e3432"
    data-rac=""
  >
    <label
      class="react-aria-Label"
      for="react-aria-«ru»"
      id="react-aria-«rv»"
    >
      Label
    </label>
    <input
      aria-labelledby="react-aria-«rv»"
      class="_input_7e3432 _input--medium_7e3432"
      data-rac=""
      id="react-aria-«ru»"
      name="snapshot"
      placeholder="Text"
      tabindex="0"
      title=""
      type="text"
      value=""
    />
  </div>
</div>
`;
